src/auth/storage.js

```diff
@@ -28,21 +28,24 @@ class Storage {
     this.store = store;
   }
 
-  get(key) { return this.store.getItem(key); }
-  set(key, value) { return this.store.setItem(key, value); }
-  remove(key) { return this.store.removeItem(key); }
-  clear() { return this.store.clear(); }
+  get(key) { return new Promise(resolve => resolve(this.store.getItem(key))); }
+  set(key, value) { return new Promise(resolve => resolve(this.store.setItem(key, value))); }
+  remove(key) { return new Promise(resolve => resolve(this.store.removeItem(key))); }
+  clear() { return new Promise(resolve => resolve(this.store.clear())); }
 }
 
-export function createStorage(type) {
-  if (type === 'localStorage') {
+export function createStorage(options) {
+  let { storageType, storage } = options;
+  if (storageType === 'localStorage') {
     if ((typeof window !== 'undefined') && 'localStorage' in window && window.localStorage !== null) {
       return new Storage(window.localStorage);
     }
-  } else if (type === 'sessionStorage') {
+  } else if (storageType === 'sessionStorage') {
     if ((typeof window !== 'undefined') && 'sessionStorage' in window && window.sessionStorage !== null) {
       return new Storage(window.sessionStorage);
     }
+  } else if (storageType == 'customStorage') { //eslint-disable-line eqeqeq
```

Comment:
why no === here?

Reply:
When I was working with `flow`, it disallowed triple equals. I'll change it back